Library/Homebrew/cmd/audit.rb

```diff
@@ -572,6 +573,18 @@ def audit_specs
     end
   end
 
+  def audit_revision
+    return unless formula.tap # skip formula not from core or any taps
+
+    fv = FormulaVersions.new(formula)
+    revision_map = fv.revision_map("origin/master")
+    if (revisions = revision_map[formula.version]).any?
+      problem "revision should not decrease" if formula.revision < revisions.max
+    else
+      problem "revision should be reset to 0" unless formula.revision == 0
```

Comment:
This should maybe be `revision should be removed`? Otherwise we will get people actually writing `revision 0`, which has happened before.